fs.go

```diff
@@ -13,6 +13,28 @@ var (
 	ErrCrossedBoundary = errors.New("chroot boundary crossed")
 )
 
+// Capability holds the supported features of a filesystem.
+type Capability uint64
```

Comment:
Documentation should be more clear about what a capability means. A capability, defined as filesystem-wide, can only be a hint and not a guarantee, because:

- Lots of filesystems, including `osfs` depending on the underlying filesystem, will have additional restrictions depending on the actual filesystem implementation, operating system, mount options, etc.
- In some cases, even if it is theoretically possible to know that a filesystem is fully read-only (e.g. `os` filesystem with read-only mount), we just do not check it.

Users should always be prepared to handle gracefully any possible error and do not assume that a capability makes any function returning an `error` always returning `nil`.